### PR TITLE
feat: `Bun.build` -> `tsdown(rolldown)`

### PR DESCRIPTION
**Pull request by ryoppippi**
Hi!
I'm the author of [bun-plugin-isolated-decl](https://github.com/ryoppippi/bun-plugin-isolated-decl)
I recently tried `tsdown` and it is better than `Bun.build`.
So this is just an introduction to you.
If you do not want to use it, just close this PR!

### CODE REVIEW

**Comment by RobinTail** [comment]
Thank you @ryoppippi ,

I'll certainly try it.

**Comment by RobinTail** [comment]
Is it better than `tsup`, @ryoppippi ?

File: package.json [hunk 1]
```diff
@@ -51,9 +50,8 @@
     "@tsconfig/bun": "^1.0.7",
     "@tsconfig/node18": "^18.2.4",
```
Comment (RobinTail) [inline package.json:51]:
```suggestion
```

this one no longer needed in this case

**Comment by ryoppippi** [comment]
@RobinTail For speed, yes.
tsup is build on top of esbuild, and esbuild's output is not good in my opinion

File: package.json (hunk 1 above)
Comment (ryoppippi) [reply package.json:51]: nice

**Comment by RobinTail** [comment]
### CJS diff

image

### ESM diff

image

**Comment by ryoppippi** [comment]
Oh, yeah mangling is not good. Do you need it?

**Comment by ryoppippi** [comment]
I think the output is similar to from 'rollup', and it is better than from 'Bun.build'

**Comment by RobinTail** [comment]
> Oh, yeah mangling is not good. Do you need it?

Just exploring how exactly it affects the output.
No action needed.
The CI only fails because of the fork.

**Comment by RobinTail** [comment]
I actually like the ESM output, @ryoppippi 
Specifically the usage of `const` over `let` (produced by both `esbuild` and `bun`).

I'm having some doubts regarding CJS:
- `use strict` is gone
- `__esModule` property is gone

I need to figure out how critical is that and which side effects it could lead to.

**Comment by ryoppippi** [comment]
In this library, `__esModule` is nothing, so you can ignore it

**Comment by ryoppippi** [comment]
maybe related to `esModuleInterop` option in tsonfig.json

**Comment by RobinTail** [comment]
yes, it's only required for default exports, that I intentionally removed earlier.
I think it's good.

**Comment by ryoppippi** [comment]
I have been experimenting with various bundlers over the past few months and have come to the conclusion that rollup is currently the best way to create a library where tree-shaking works well. esbuild and bun are lax with tree-shaking.

We think roldown is very good because it is as close as possible to the output of rollup and it works fast. (Except that JSDoc disappears from dts)

**Comment by RobinTail** [comment]
Thank you very much, @ryoppippi 
I highly appreciate your help and the suggestion!

**Comment by ryoppippi** [comment]
@RobinTail Also, if you are interested in, check out my new cli tool for Bun
https://github.com/ryoppippi/bunpare

**Comment by RobinTail** [comment]
mmm, that's actually a very useful feature, @ryoppippi !
thank you!

**Pull request merged**